File: src/main/java/com/wishai/xzrtw/config/XzrtPropertiesFactory.java
package com.wishai.xzrtw.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@EnableConfigurationProperties(XzrtProperties.class)
public class XzrtPropertiesFactory {

    private static XzrtProperties xzrtProperties;

    @Autowired
    private XzrtPropertiesFactory(XzrtProperties xzrtProperties) {
        XzrtPropertiesFactory.xzrtProperties = xzrtProperties;
    }

    public static XzrtProperties getXzrtProperties() {
        return xzrtProperties;
    }

}

File: src/main/java/com/wishai/xzrtw/config/XzrtPropertiesFactory.kt
package com.wishai.xzrtw.config

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.stereotype.Component

@Component("XzrtPropertiesFactory")
@EnableConfigurationProperties(XzrtProperties::class)
class XzrtPropertiesFactory
@Autowired
constructor(xzrtProperties: XzrtProperties) {

    init {
        XzrtPropertiesFactory.xzrtProperties = xzrtProperties
    }

    companion object {
        var xzrtProperties: XzrtProperties? = null
            private set
    }

}
